### PR TITLE
Fix joinQueue bugs

### PR DESCRIPTION
**Pull request by yonbergman**
- fixed joining the queue as first player
- fixed trying to join queue while already in active match
- Cleanup of cloud code
- https://github.com/iic-ninjas/mokojin/issues/51

### CODE REVIEW

**Comment by udiWertheimer** [comment]
A queue is born

File: cloud/main.js [hunk 1]
```diff
@@ -67,39 +67,26 @@ Parse.Cloud.define("joinQueue", function(request, response) {
...
-  )
+  Person.find(personId).then(function(person){
+    if (!person) return Parse.Promise.error("No person matching id");
+    return person.joinQueue();
+  }).then(function(player){
+    if (!player) return Parse.Promise.error("No player added");
+    return Match.currentMatch().then(function(match){
+      if (match) return match;
+      return Match.startMatch().fail(function(err){
+        return Parse.Promise.as(player);
```
Comment (giladgo) [inline cloud/main.js:78]: just `return player`

**Comment by giladgo** [comment]
:rabbit:

File: cloud/main.js (hunk 1 above)
Comment (yonbergman) [inline cloud/main.js:78]: doing that didn't reset the error to success

**Pull request merged**